src/bentoml/server.py

```diff
@@ -40,7 +42,7 @@ class Server(ABC):
 
     def __init__(
         self,
-        bento: str | Bento | Tag,
+        bento: str | Bento | Tag | Service,
```

Comment:
Will we need bentofile.yaml in some cases now or in the future, because Service + bentofile.yaml = Bento.

Reply:
currently, we only use `bentofile.yaml` for getting the service instance when running `bentoml serve` in a folder containing a `bentofile.yaml`. The problem with adding `bentofile.yaml` path here in this API is that we can not guarantee the working directory is the same as the current Python process. I feel for Python API, the most intuitive usage is still importing the service object directly, instead of using a string like "service.py:svc" or passing in the "bentofile.yaml"

Reply:
That makes sense. Adding this API will also limit us from using data in bentofile.yaml during `serve`. We don't use it today but that's an assumption made with `bentoml serve`.